data-analytics/iot-nirvana/app-engine/src/main/java/com/google/cloud/demo/iot/nirvana/servlet/rest/task/gce/InstanceCreation.java

```diff
@@ -0,0 +1,182 @@
+    instanceName = baseInstanceName + "-" + UUID.randomUUID().toString();
+
+    /** DISK */
+    // define disks
+    AttachedDiskInitializeParams newDiskInitParams = new AttachedDiskInitializeParams();
+    newDiskInitParams.setDiskName("disk-" + instanceId);
+    newDiskInitParams.setSourceImage(Constants.GCE_LINUX_JAVA);
+
+    AttachedDisk newDisk = new AttachedDisk();
+    newDisk.setBoot(true); // boot the instance using this disk
+    newDisk.setAutoDelete(true); // delete the disk when the associated instance will be deleted
+    newDisk.setInitializeParams(newDiskInitParams);
+
+    List<AttachedDisk> disks = new ArrayList<AttachedDisk>();
+    disks.add(newDisk);
+
+    /** NETWORK */
+    // define network
+    NetworkInterface net = new NetworkInterface();
+    net.setNetwork(Constants.GCE_NETWORK_TYPE);
+    // define access config (only needed for public IP)
+    AccessConfig ac = new AccessConfig();
+    ac.setType(Constants.GCE_NETWORK_EMPHERAL);
+    List<AccessConfig> acs = new ArrayList<AccessConfig>();
+    acs.add(ac);
+    net.setAccessConfigs(acs);
+    List<NetworkInterface> networks = new ArrayList<NetworkInterface>();
+    networks.add(net);
+
+    /** METADATA STARTUP SCRIPT */
+    // Define metadata items
+
+    // startup script
+    // read the path of STARTUP file
+
+    Metadata.Items itemStartupScript = new Metadata.Items();
+    itemStartupScript.setKey(Constants.GCE_METADATA_STARTUP_KEY);
+    itemStartupScript.setValue(
+        Utils.getAppEngineProperty(this.getServletContext(), "GCE_METADATA_STARTUP_VALUE"));
+    // input instance name
+    Metadata.Items instanceNumberMetadata = new Metadata.Items();
+    instanceNumberMetadata.setKey(Constants.GAE_TASK_GCE_PARAM_NAME_INSTANCE_NUMBER);
+    instanceNumberMetadata.setValue(instanceNumber);
+
+    // create array of metadata
+    List<Metadata.Items> items = new ArrayList<Metadata.Items>();
+    items.add(itemStartupScript);
+    items.add(instanceNumberMetadata);
+    Metadata metadata = new Metadata();
+    metadata.setItems(items);
+
+    // define Service Account to be used in the GCE
+    // (for a sake of simplicity a full-scopes scope has been used)
+    ServiceAccount account = new ServiceAccount();
+    account.setEmail("default");
+    List<String> scopes = new ArrayList<>();
+    scopes.add("https://www.googleapis.com/auth/cloud-platform");
+    account.setScopes(scopes);
+
+    /** INSTANCE definition */
+    Instance newInstance = new Instance();
+    newInstance.setName(instanceName);
+    newInstance.setMachineType(
+        Constants.GCE_MACHINE_TYPE_FULL_URL
+            .replaceAll("xxxZONExxx", instanceZone)
+            .replaceAll("xxxMACHINE_TYPExxx", instanceType));
+    newInstance.setDisks(disks);
+    newInstance.setNetworkInterfaces(networks);
+    newInstance.setMetadata(metadata);
+    newInstance.setServiceAccounts(Collections.singletonList(account));
+
+    /** BOOTSTRAP machine */
+    int retry = 0;
```

Comment:
LIne 164 overwrites the initialization. we can declare retry =5

Reply:
The declaration retry = 5 at line 164 was used to exit the while{} loop after the VM instance was successfully launched. Changed the condition in commit https://github.com/GoogleCloudPlatform/professional-services/pull/61/commits/b1e730501178e94d56abd659207be0869861d92c.